watertap/examples/flowsheets/mvc/mvc_single_stage_ui.py

```diff
@@ -0,0 +1,649 @@
+    )
+    exports.add(
+        obj=fs.hx_brine.costing.capital_cost,
+        name="Brine heat exchanger",
+        ui_units=fs.costing.base_currency,
+        display_units="$",
+        rounding=0,
+        description="Brine heat exchanger capital cost",
+        is_input=False,
+        is_output=True,
+        output_category="Capital costs",
+    )
+    exports.add(
+        obj=fs.mixer_feed.costing.capital_cost,
+        name="Feed mixer",
+        ui_units=fs.costing.base_currency,
+        display_units="$",
+        rounding=0,
+        description="Feed mixer capital cost",
+        is_input=False,
+        is_output=True,
+        output_category="Capital costs",
+    )
+    exports.add(
+        obj=fs.evaporator.costing.capital_cost,
+        name="Evaporator",
+        ui_units=fs.costing.base_currency,
+        display_units="$",
+        rounding=0,
+        description="Evaporator capital cost",
+        is_input=False,
+        is_output=True,
+        output_category="Capital costs",
+    )
+    exports.add(
+        obj=fs.compressor.costing.capital_cost,
+        name="Compressor",
+        ui_units=fs.costing.base_currency,
+        display_units="$",
+        rounding=0,
+        description="Compressor capital cost",
+        is_input=False,
+        is_output=True,
+        output_category="Capital costs",
+    )
+    # Operating costs
+    exports.add(
+        obj=fs.costing.total_operating_cost,
+        name="Total",
+        ui_units=fs.costing.base_currency / pyunits.year,
+        display_units="$/year",
+        rounding=0,
+        description="Total annual operating costs - including electricity, heating, "
+        "and fixed",
+        is_input=False,
+        is_output=True,
+        output_category="Operating costs",
+    )
+    exports.add(
+        obj=fs.costing.aggregate_flow_costs["electricity"],
+        name="Electricity",
+        ui_units=fs.costing.base_currency / pyunits.year,
+        display_units="$/year",
+        rounding=0,
+        description="Annual electricity costs ",
+        is_input=False,
+        is_output=True,
+        output_category="Operating costs",
+    )
+
+
+def build_flowsheet(build_options=None, **kwargs):
+    # build and solve initial flowsheet
+    m = build()
+    set_operating_conditions(m)
+    add_Q_ext(m, time_point=m.fs.config.time)
+    initialize_system(m)
+    # rescale costs after initialization because scaling depends on flow rates
+    scale_costs(m)
+    fix_outlet_pressures(m)  # outlet pressure are initially unfixed for initialization
+
+    m.fs.objective = Objective(expr=m.fs.Q_ext[0])
+
+    solver = get_solver()
+    results = solve(m, solver=solver, tee=False)
+    assert_optimal_termination(results)
+
+    m.fs.Q_ext[0].fix(0)  # no longer want external heating in evaporator
+    del m.fs.objective
+    set_up_optimization(m)
+    results = solve(m, solver=solver, tee=False)
+
+    return m
+
+
+def solve_flowsheet(flowsheet=None):
```

Comment:
```suggestion
def solve_flowsheet(flowsheet):
```
Not sure why this would be an optional argument.

Reply:
I'm not sure of the logic behind it either, but it seems like all of the ui files do this. @dangunter or @MichaelPesce Would there be any problems with making this change?

Reply:
I see no reason why this would be a problem